### DISCUSSION TITLE
Delete button

### DISCUSSION THREAD

**Discussion by alucard29** [saade/filament-fullcalendar#62] [Ideas]
Hi !
Could it be possible to add a delete button on the edit modal ?
Regards

**Comment by Lucathiel** [comment]
Hello @alucard29,

Don't know if you managed to do what you want since then but here is what i do to perform the delete action, that could be useful for somebody else : 

1. Publish views with the following command
```
php artisan vendor:publish --tag="filament-fullcalendar-views"
```

2. Edit the new file in your views _edit-event.modal.blade.php_ by adding a simple button that listen to an event in the modal
```
<button  x-on:click="Livewire.emit('deleteEvent')">
     Supprimer
</button>
```

3. Add the following listener and method in your class extending FullCalendarWidget
```
protected $listeners = [
    'deleteEvent' => 'deleteEvent',
];

public function deleteEvent()
{
    $this->event->delete();

    $this->refreshEvents();

    $this->dispatchBrowserEvent('close-modal', ['id' => 'fullcalendar--edit-event-modal']);
}
```

Ideally, it should be done properly in a PR with a validation modal to perform it but this is a _quick local fix_

**Reply by alucard29** [reply]
Thnks it works like a charm !

**Comment by saade** [comment]
Hi! I would accept a PR for adding this to the core. Feel free to PR it.
